### ISSUE TITLE
Interface: allow specifying certain basic blocks as cold

### ISSUE THREAD

**Issue by Amanieu** [bnjbvr/minira#13]
Basic block frequencies are currently estimated using loop depth, but sometimes the front-end knows certain code paths are almost never taken (e.g. error handling). It would be useful to be able to communicate this to the register allocator.

**Comment by julian-seward1** [comment]
Yes.  (I agree).  The current loop-depth code doesn't even take into account if-then-else effects on frequency -- it was just intended as a the simplest thing we could do, that would drive the new allocator work.  It seems to me very likely that indeed, the interface will at some point be extended to allow cold-path info to the allocator.
